### DISCUSSION TITLE
How long support of Adobe air with HTMLoader style i.c.m webkit?

### DISCUSSION THREAD

**Discussion by mzonderland** [airsdk/Adobe-Runtime-Support#2483] [General]
I would like to discuss a problem we have in combination with our application and a 3D Viewer and Adobe AIR.
From our application it is possible to execute a function which starts the 3D viewer.

The latest AIR runtimes do not include a custom build of WebKit, instead they use a WebView style component.
And since 33.1.1.932 the HTMLLoader build-in webkit is removed and native StageWebView is used instead.
This I found in the Harman release notes.

Our application relies on a HTMLLoader to open and control the viewer .
And we can use the AIR runtime including WebKit - version 33.1.1.744 which works great and is found here:
https://airsdk.harman.com/runtime

Is there anybode who knows how long Harman continue to support the older HTMLoader style with webkit?

**Comment by ajwfrost** [comment]
That depends a bit on what you mean by "support"... we have got a branch of the code based on 33.1.1.743, from which we can do updates (such as the 33.1.1.744 version that you mention). So if there are any really crucial issues that we've fixed from the "main" set of releases that would be essential for this branch, we can look at integrating those on a case-by-case basis. But we're not planning for that to include any feature updates or AS3 API changes, so it's more a case of the 33.1.1.743+ versions just being made available for those folk who really need the WebKit capabilities and don't mind the restrictions on macOS with Apple silicon.

But in terms of maintaining the 33.1.1.744 and later versions on that branch, this shouldn't be a particular problem to continue to make these available for those who need them.

We looked at getting another browser engine integrated, where the other engine could then be separately maintained but would allow for some of the similar communications/interactions between JS and AS3, but the candidates we found would all need licensing.. we're still looking to enable some hooks into the rendering flow so that this could be done via an ANE, which might allow folk to recreate a similar mechanism (without then being dependent upon us to maintain/update the browser engine).

thanks
